.github/workflows/ci-windows.yml

```diff
@@ -49,18 +47,15 @@ jobs:
           
       - name: "Checking out repository"
         env:
-          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+          BRANCH_NAME: ${{ github.ref }}
           REPO_ADDRESS: ${{ github.server_url }}/${{ github.repository }}
+        shell: bash
         run: |
-          git \
-            -c submodule."third_party/torch-mlir".update=none \
-            -c submodule."third_party/stablehlo".update=none \
-            -c submodule."src/runtime_src/core/common/aiebu".update=none \
-            clone \
-            --recursive \
-            --shallow-submodules \
-            --depth 1 \
-            -b $BRANCH_NAME $REPO_ADDRESS .
+          git init
+          git remote add origin $REPO_ADDRESS
+          git -c protocol.version=2 fetch --depth 1 origin $BRANCH_NAME
+          git reset --hard FETCH_HEAD
+          git -c submodule."third_party/torch-mlir".update=none -c submodule."third_party/stablehlo".update=none -c submodule."src/runtime_src/core/common/aiebu".update=none submodule update --init --recursive --depth 1 --single-branch -j 10
```

Comment:
Another option for this style of git operations is to use a script like https://github.com/iree-org/iree/blob/main/build_tools/scripts/git/update_runtime_submodules.sh. Sample usage here: https://github.com/iree-org/iree/blob/45132f5b1f1697e1f4d3450f0d6371c2ec49700a/.github/workflows/ci.yml#L209-L213. That does require checking out the repo with the script (without submodules) though.

With such a script, developers and workflows can share the same steps without copy/pasting all this.

Reply:
in general i'm not a fan of hiding stuff in scripts that get used in CI ymls - at least until github let's you jump to those scripts like it does for names in cpp. the exception (at least for iree-amd-aie) being cmake configure/build, we would put into a script.

here in particular, this incantation isn't useful for devs because it's overly complex because it needs to work for forks (which i guess you well know from that example repo you showed me).